gremlin-python/src/main/python/gremlin_python/driver/client.py

```diff
@@ -154,20 +155,19 @@ def submit_async(self, message, bindings=None, request_options=None):
 
         log.debug("message '%s'", str(message))
         fields = {'g': self._traversal_source}
-        if isinstance(message, traversal.Bytecode):
-            fields['gremlinType'] = 'bytecode'
-        elif isinstance(message, str):
-            fields['gremlinType'] = 'eval'
 
         if isinstance(message, str) and bindings:
             fields['bindings'] = bindings
 
-        if isinstance(message, traversal.Bytecode) or isinstance(message, str):
+        if isinstance(message, str):
             log.debug("fields='%s', gremlin='%s'", str(fields), str(message))
             message = request.RequestMessageV4(fields=fields, gremlin=message)
 
         conn = self._pool.get(True)
         if request_options:
-            message.fields.update(request_options)
+            message.fields.update({token: request_options[token] for token in TokensV4
+                                   if token in request_options})
+            if 'params' in request_options:
```

Comment:
Is there a distinction between `bindings` and `request_options['params']`? Do we need to have both or can we simplify this interface a bit?

If we do keep both, would it be better to merge both dicts together instead of taking either `bindings` and `request_options['params']`?

Reply:
I'm fairly certain that bindings should be sent within request_options, but python has been having it separated. I'm not sure if we should change the API with this PR yet, so I've updated the `request_options['params']` to add to the `bindings` dict instead.

Reply:
good to have test for case when both `bindings` and request_options are present